server/src/main/java/org/elasticsearch/index/engine/NoOpEngine.java

```diff
@@ -116,4 +124,52 @@ public SegmentsStats segmentsStats(boolean includeSegmentFileSizes, boolean incl
             return super.segmentsStats(includeSegmentFileSizes, includeUnloadedSegments);
         }
     }
+
+    /**
+     * This implementation will trim existing translog files using a {@link TranslogDeletionPolicy}
+     * that retains nothing but the last translog generation from safe commit.
+     */
+    @Override
+    public void trimUnreferencedTranslogFiles() {
+        final Store store = this.engineConfig.getStore();
+        store.incRef();
```

Comment:
We might not need to incRef of the store here for we [already](https://github.com/elastic/elasticsearch/blob/05689de55f70736aed4d556f847e3112c99cc8a3/server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java#L101) do it in the constructor of ReadOnlyEngine.

Reply:
I added this to prevent the store to be closed while it is used to list commits... I still think that this can happen if the shard is closed?

Reply:
I might miss something here. When we create a ReadOnlyEngine, we increase the store reference and only decrease that reference in the destructor of ReadOnlyEngine. If the engine was closed, then `ensureOpen` would throw AlreadyClosedException. I am fine with your extra protection here.

Reply:
:+1: